modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumer.scala

```diff
@@ -0,0 +1,42 @@
+package ch.epfl.bluebrain.nexus.kg.service.queue
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.kafka.ConsumerMessage.CommittableOffsetBatch
+import akka.kafka.scaladsl.Consumer
+import akka.kafka.{ConsumerSettings, Subscriptions}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import io.circe.Decoder
+import io.circe.parser._
+import journal.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object KafkaConsumer {
+
+  private val log = Logger[this.type]
+
+  final def start[T](consumerSettings: ConsumerSettings[String, String],
+                     index: T => Future[Unit],
+                     topic: String)(implicit as: ActorSystem, ec: ExecutionContext, D: Decoder[T]): Future[Done] = {
+    implicit val mt: ActorMaterializer = ActorMaterializer()
+
+    Consumer
```

Comment:
About the `offset`, what happens when we restart the service, where does this offset get stored? I don't see anywhere that we are using the `ResumableProjection` to store the offset in a Cassandra table